lib/constants.py

```diff
@@ -71,4 +100,80 @@ def block_to_number(blockname: str) -> int:
+  vs = Sa2SeedKey(simos18_sa2_script, int.from_bytes(seed, "big"))
+  return vs.execute().to_bytes(4, 'big')
+
+data_records : List[DataRecord] = [
+ DataRecord(0xF190, 0, "VIN Vehicle Identification Number"),
+ DataRecord(0xF19E, 0, "ASAM/ODX File Identifier"),
+ DataRecord(0xF1A2, 0, "ASAM/ODX File Version"),
+ DataRecord(0xF40D, 1, "Vehicle Speed"),
+ DataRecord(0xF806, 1, "Calibration Verification Numbers"),
+ DataRecord(0xF187, 0, "VW Spare Part Number"),
+ DataRecord(0xF189, 0, "VW Application Software Version Number"),
+ DataRecord(0xF191, 0, "VW ECU Hardware Number"),
+ DataRecord(0xF1A3, 0, "VW ECU Hardware Version Number"),
+ DataRecord(0xF197,0,"VW System Name Or Engine Type"),
+ DataRecord(0xF1AD,0,"Engine Code Letters"),
+ DataRecord(0xF1AA,0,"VW Workshop System Name"),
+ DataRecord(0x0405,1,"State Of Flash Memory"),
+ DataRecord(0x0407,1,"VW Logical Software Block Counter Of Programming Attempts"),
+ DataRecord(0x0408,1,"VW Logical Software Block Counter Of Successful Programming Attempts"),
+ DataRecord(0x0600,1,"VW Coding Value"),
+ DataRecord(0xF186,1,"Active Diagnostic Session"),
+ DataRecord(0xF18C,0,"ECU Serial Number"),
+ DataRecord(0xF17C,0,"VW FAZIT Identification String"),
+ DataRecord(0xF442,1,"Control Module Voltage"),
+ DataRecord(0xEF90,1,"Immobilizer Status SHE"),
+ DataRecord(0xF1F4,0,"Boot Loader Identification"),
+ DataRecord(0xF1DF,1,"ECU Programming Information"),
+ DataRecord(0xF1F1,1,"Tuning Protection SO2"),
+ DataRecord(0xF1E0,1,""),
+ DataRecord(0x12FC,1,""),
+ DataRecord(0x12FF,1,""),
+ DataRecord(0xFD52,1,""),
+ DataRecord(0xFD83,1,""),
+ DataRecord(0xFDFA,1,""),
+ DataRecord(0xFDFC,1,""),
+ DataRecord(0x295A,1,"Vehicle Mileage"),
+ DataRecord(0x295B,1,"Control Module Mileage"),
+ DataRecord(0xF190,0,"VIN Vehicle Identification Number"),
+ DataRecord(0xF19E,0,"ASAM/ODX File Identifier"),
+ DataRecord(0xF1A2,0,"ASAM/ODX File Version"),
+ DataRecord(0xF15B,1,"Fingerprint and Programming Date"),
+ DataRecord(0xF191,0,"VW ECU Hardware Number"),
+ DataRecord(0xF1A3,0,"VW ECU Hardware Version Number"),
+ DataRecord(0xF187,0,"VW Spare Part Number"),
+ DataRecord(0xF189,0,"VW Application Software Version Number"),
+ DataRecord(0xF1F4,0,"Boot Loader Identification"),
+ DataRecord(0xF197,0,"VW System Name Or Engine Type"),
+ DataRecord(0xF1AD,0,"Engine Code Letters"),
+ DataRecord(0xF17C,0,"VW FAZIT Identification String"),
+ DataRecord(0xF1A5,1,"VW Coding Repair Shop Code Or Serial Number (Coding Fingerprint),"),
+ DataRecord(0x0405,1,"State Of Flash Memory"),
+ DataRecord(0xF1AB,0,"VW Logical Software Block Version"),
+ DataRecord(0xF804,0,"Calibration ID"),
+ DataRecord(0xF17E,0,"ECU Production Change Number")
+]
+
+# When we're performing WriteWithoutErase, we need to write 8 bytes at a time in "patch areas" to allow the ECC operation to be performed correctly across the patched data.
```

Comment:
Note to self: I'd like these comments to live in the flasher, to keep the flasher documentation all with the flash process. 

However, it makes sense for the transfer sizes to live here. Really, these transfer sizes should live in a config file that's "paired" with each patch file.